### PR TITLE
** Do NOT merge yet please! ** Testing and frontend setup 

### PR DESCRIPTION
**Pull request by garrowat**
### Testing and frontend setup 

This is an intermediate pull-request to make my work visible to the team; **please do not merge yet!**

**Todo:** 

- [x] Setup Jest & Enzyme
- [x] Webpack
- [x] Babel
- [x] React, ReactDOM
- [x] Dynamic Routes
- [x] Code the components to dynamically display data
- [x] Setup JSS
- [x] Style the components
- [ ] Test components with Enzyme and Jest

### CODE REVIEW

**Comment by garrowat** [comment]
Ready for merge, so that we can all use it for proxies! I still have to implement the style slider, but the component should work for now for our purposes. Also the css is not finalized at all.

File: client/ProductDescription.jsx [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import React from 'react';
 
-const ProductDescription = props => <div>ProductDescription Here</div>;
+const ProductDescription = ({ name, price }) => (
```
Comment (matt-leyrat) [inline client/ProductDescription.jsx:3]: I'm diggin this destructuring 💯

File: client/RelatedProducts.jsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default class RelatedProducts extends React.Component {
         <span>You may also like</span>
         <ProductContainer>
           {
-            products.map(product => <Product product={product} />)
+            products.map(product => <Product key={product.id} product={product} />)
```
Comment (matt-leyrat) [inline client/RelatedProducts.jsx:60]: Should the id's be more unique? You noticed it on my PR so I figured I'd mention it as well :)

**Review by matt-leyrat — COMMENTED** [review]
LGTM, easy to read and elegant use of ES6  👍

File: client/RelatedProducts.jsx (hunk 2 above)
Comment (garrowat) [reply client/RelatedProducts.jsx:60]: Yes! Nice catch

File: client/RelatedProducts.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const pid = window.location.pathname.substring(4);
+    for (let i = 0; i < 4; i += 1) {
+      const currentPid = parseInt(pid, 10) + (i * 4);
+      console.log(currentPid);
+      this.getProduct(currentPid);
+    }
+  }
+
+  getProduct(pid) {
+    fetch(`http://127.0.0.1:3003/api/${pid}`)
```
Comment (garrowat) [inline client/RelatedProducts.jsx:44]: Realizing now that I totally have to set up relative urls for the api calls

**Pull request merged**